Cleaned Up Build System File Tracking and Other Minor Fixes

jansky opened the pull request:
Based on the fixes made by @bbkc22113 in #8, I have made some additional changes as suggested by @TheMoreLaw:

> @jansky: As for the build system, if certain files are generated based on configs, IMO, these should not be tracked. They should be regenerated as part of the install script. Please untrack them, and add these files to the .gitignore file.

I have untracked all files generated by autotools when running the autobuild script.

```
autobuild.sh:
# Checking for Operating System
if [ "$(uname)" == "Darwin" ]; then
# Install dependencies and Generate Makefile on Mac
chmod a+x scripts/prepare_env_mac.sh && scripts/prepare_env_mac.sh
bash scripts/build_mac.sh
```
> Why change this?
> I suppose my question was more around using the bash command vs the chmod and then executing.

I was worried that there were some bash-specific features used in the shell scripts. However, after a few modifications I have made sure that the autobuild scripts works with bash and plain sh, so I've removed this.

```
README.md:
In case you run into problems while building the generator using the automated script, you could try manually carrying out some of the steps.

#### macOS
1. Install fftw and gcc.

- brew install fftw && brew link fftw
- brew install gcc
+ $ brew install fftw
```
> Why change this?

I've reverted this change.

```
README.md:
3. The `cartogram` executable can be found under `cartogram_generator/`. To install it, run the following command:
```
> Why not copy it to the root directory, as before? Is there a need to change this?

I've changed the autobuild script to copy the executable like before.

bernardboey committed:
Updated generator, build system, readme, sample data and images

bernardboey committed:
Updated README

Commit:
build system fixes

Commit:
removed aclocal.m4

pratyushmore commented:
@jansky based on this diff, the `.deps` folder is still tracked. Can you please elaborate. Also, I would recommend adding this folder to the `.gitignore` file. 

Based on your earlier response, you needed the following 3 files for the build system:
`configure.ac`, `Makefile.am`, `cartogram_generator/Makefile.am`.

Yet, I see many more files in this diff. Am I missing something?

Also, there are a few comments from the other PR regarding the README that are unaddressed. Please also address them before I approve this.

pratyushmore commented on README.md:
```diff
@@ -41,34 +53,25 @@ Gastner, M., Seguy, V., & More, P. (2018). Fast flow-based algorithm for creatin
-2. a `.dat` file containing the data (such as population) for each region, according to which these will be scaled. For the 2016 US presidential election data we provide `usa_electors.dat` (also in the `sample_data` folder) which provides the number of electors for each state.
-
-<!-- We can add some explanation about the files here if needed -->
+The cartogram generator expects two input file.
```
two input files.

jansky replied:
Fixed in 4287467.

pratyushmore commented on README.md:
````diff
@@ -84,41 +87,55 @@ No additional dependencies. Your default `apt-get` package manager should work f
+sudo ./autobuild.sh
+```
+
+3. The `cartogram` executable can be found under `cartogram_generator/`. To install it, run the following command:
````
Is it still under `cartogram_generator/`? I thought you are now copying it to root?

Also, the meaning of `install` is unclear. Maybe something along the lines of: `To add the generator to your list of binaries, and access it outside this directory,  please run the following command:`

jansky replied:
Fixed in 4287467.

pratyushmore commented on README.md:
```diff
@@ -84,41 +87,55 @@ No additional dependencies. Your default `apt-get` package manager should work f
-**Note:** In our repository, we also include data for India and China's GDP, segmented by their states/provinces.
-
-To generate the India GDP cartogram, run:
+The cartogram will generate 2 new files: a processedmap '.json' file and '.csv' file. Fill in the data in the '.csv' file. Then, run the following command to generate the cartogram.
```
What does `fill in the data in the .csv` file mean?

jansky replied:
I've added more specific instructions in 4287467.

pratyushmore commented on README.md:
````diff
@@ -84,41 +87,55 @@ No additional dependencies. Your default `apt-get` package manager should work f
 ```
 
-3. You should see two generated files - `map.eps` showing the original map, and `cartogram.eps` showing the generated cartogram.
+If you supply the **-e** option, you should see two generated files:
````
You should add the option in the example itself, if you are referencing it. Explain why you added the option.

jansky replied:
Fixed in 4287467.

pratyushmore commented on README.md:
````diff
@@ -130,28 +147,40 @@ evince <filename>.eps
-```
-FATAL:/opt/local/bin/../libexec/as/x86_64/as: I don't understand 'm' flag!
-```
+**-e**<br/>
````
Please explain this option. Also, should the default be to generate the EPS, and there could be an option to not generate it?

jansky replied:
I've added more explanation of this option in 4287467. I'd rather not change the default behavior of the application, because this would mean updating the go-cart web application.

pratyushmore reviewed (commented):
Adding some remaining issues into this MR.

Commit:
untracked more build system files

Commit:
updated README

Commit:
updated shell scripts

jansky commented:
I've removed additional files generated by the build system that shouldn't be tracked. I have also addressed the comments above, and reworked the README sections about how to use the cartogram generator.

pratyushmore reviewed (approved):
@jansky Thanks for addressing the comments. This is a great contribution: enabling users to input modern data formats, thus making this software more accessible to the community!
